src/Simplex_tree/test/simplex_tree_make_filtration_non_decreasing_unit_test.cpp

```diff
@@ -0,0 +1,148 @@
+  
+  // Copy original simplex tree
+  typeST st_copy = st;
+
+  // Modify specific values for st to become like st_copy thanks to make_filtration_non_decreasing
+  st.assign_filtration(st.find({0,1,6,7}), 0.8);
+  st.assign_filtration(st.find({0,1,6}), 0.9);
+  st.assign_filtration(st.find({0,6}), 0.6);
+  st.assign_filtration(st.find({3,4,5}), 1.2);
+  st.assign_filtration(st.find({3,4}), 1.1);
+  st.assign_filtration(st.find({4,5}), 1.99);
+  
+  std::cout << "Check the simplex_tree is rolled back in case of decreasing filtration values" << std::endl;
+  BOOST_CHECK(st.make_filtration_non_decreasing());
+  BOOST_CHECK(st == st_copy);
+
+  // Other simplex tree
+  typeST st_other;
+  st_other.insert_simplex_and_subfaces({2, 1, 0}, 3.0);  // This one is different from st
+  st_other.insert_simplex_and_subfaces({3, 0}, 2.0);
+  st_other.insert_simplex_and_subfaces({3, 4, 5}, 2.0);
+  st_other.insert_simplex_and_subfaces({0, 1, 6, 7}, 1.0);
+
+  // Modify specific values for st to become like st_other thanks to make_filtration_non_decreasing
+  st.assign_filtration(st.find({2}), 3.0);
+  // By modifying just the simplex {2}
+  // {0,1,2}, {1,2} and {0,2} will be modified
+  
+  std::cout << "Check the simplex_tree is repaired in case of decreasing filtration values" << std::endl;
+  BOOST_CHECK(st.make_filtration_non_decreasing());
+  BOOST_CHECK(st == st_other);
+
+  // Modify specific values for st still to be non-decreasing
+  st.assign_filtration(st.find({0,1,2}), 10.0);
+  st.assign_filtration(st.find({0,2}), 9.0);
+  st.assign_filtration(st.find({0,1,6,7}), 50.0);
+  st.assign_filtration(st.find({0,1,6}), 49.0);
+  st.assign_filtration(st.find({0,1,7}), 48.0);
+  // Other copy simplex tree
+  typeST st_other_copy = st;
+  
+  std::cout << "Check the simplex_tree is not modified in case of non-decreasing filtration values" << std::endl;
+  BOOST_CHECK(!st.make_filtration_non_decreasing());
+  BOOST_CHECK(st == st_other_copy);
+  
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(make_filtration_non_decreasing_on_nan_values, typeST, list_of_tested_variants) {
+  typeST st;
+
+  st.insert_simplex_and_subfaces({2, 1, 0}, std::numeric_limits<double>::quiet_NaN());
+  st.insert_simplex_and_subfaces({3, 0},    std::numeric_limits<double>::quiet_NaN());
+  st.insert_simplex_and_subfaces({3, 4, 5}, std::numeric_limits<double>::quiet_NaN());
+  
+  /* Inserted simplex:     */
+  /*    1                  */
+  /*    o                  */
+  /*   /X\                 */
+  /*  o---o---o---o        */
+  /*  2   0   3\X/4        */
+  /*            o          */
+  /*            5          */
+
+  std::cout << "SPECIFIC CASE:" << std::endl;
+  std::cout << "Insertion with NaN values does not ensure the filtration values are non decreasing" << std::endl;
+  st.make_filtration_non_decreasing();
+
+  std::cout << "Check all filtration values are NaN" << std::endl;
+  for (auto f_simplex : st.filtration_simplex_range()) {
```

Comment:
Why not complex_simplex_range? filtration_simplex_range is a more costly, more specific tool when you need the ordering.

Reply:
You are right. Fixed on 19ea0c1.
complex_simplex_range is not binded. How could we call it ? We have `get_skeleton` and `get_filtration`. `get_simplices` ? May be confusing as `get_filtration` returns simplex and its filtration value ?

Reply:
> Fixed on [19ea0c1](https://github.com/GUDHI/gudhi-devel/commit/19ea0c10f283188282a78ebebf4c1a51f2f40040).

Thanks.

> `get_simplices` ?

Sure.

> May be confusing as `get_filtration` returns simplex and its filtration value ?

In some sense, it is slightly confusing that get_skeleton returns the filtration values. But if we consider that simplices always come with their filtration value, then get_simplices could be equivalent to get_skeleton(max), i.e. return the same objects as get_filtration, but without a specified order, and thus more cheaply (at least if you only iterate once), without initialize_filtration. Or we could have get_simplices to get just the simplices, and get_simplices_with_filtration to get the pairs. Or some other variant. We should discuss this elsewhere though, it isn't related to this PR anymore...